central/graphql/generator/codegen/utils.go

```diff
@@ -41,6 +41,12 @@ func plural(s string) string {
 }
 
 func importedName(p reflect.Type) string {
+	// Handle collision of "v1":
+	// "github.com/stackrox/rox/generated/api/v1"
+	// "github.com/stackrox/scanner/generated/scanner/api/v1"
+	if strings.HasSuffix(p.PkgPath(), "scanner/generated/scanner/api/v1") {
+		return fmt.Sprintf("scannerV1.%s", p.Name())
```

Comment:
Could you elaborate a bit how this is used?
I tracked it back to codegen.go, but I think I'm missing a bit of knowledge about graphql and/or our code generation involved here.

Reply:
This is used in `make generated-srcs` in one of the steps that can be executed isolated by:  

```
go run central/graphql/resolvers/gen/main.go
mv generated.go central/graphql/resolvers/generated.go
```

When you run this, then the generated code will refer to some libraries using `v1.FuncName`. Before this PR, the `v1` was `github.com/stackrox/rox/generated/api/v1` but now we start using also `github.com/stackrox/scanner/generated/scanner/api/v1` so using the `v1`  would be ambiguous.

Thus, I proposed that all imports ending with  `scanner/generated/scanner/api/v1` shall be addressed by a named import `scannerV1`, while `github.com/stackrox/rox/generated/api/v1` would remain the same.


By introducing `message Components {}` locally inside `NodeInventory` (Instead of using `scannerV1.Components components` I minimalized the number of conflicts between `v1` and `scannerv1`, but that change still has influence on the generated code.

If we would keep `scannerV1.Components components` in the `NodeInventory` in `node.proto`, we would be forced to solve multiple other conflicts concerning functions named idetically but having different types of parameters. In that case, the change in `[codegen/utils.go](https://github.com/stackrox/stackrox/pull/3757/files/3dbf09e0d79e1392a2d4fd6733d26770966b4681#diff-47afd3b752f5e9f8f4eb339e51277b30015312ff03684e7977dc66d411e05e3f)` would not be sufficient.